### ISSUE TITLE
Prompt-to-Prompt Image Editing with Cross Attention Control

### ISSUE THREAD

**Issue by reyllama** [reyllama/paper-reviews#181]
# TL; DR

- The devil is in the cross-attention map; manipulate text-condition cross-attention map for mask-free image manipulation.

![image](https://user-images.githubusercontent.com/46308407/185853052-de1c27f6-1485-4fe7-80d7-d18eb81396b8.png)

# Motivation

- Immense success of text-to-image models (*e.g., Dall-E-2, Imagen*)
- Inconvenience of manually providing spatial mask for local image editing

# Formulation

![image](https://user-images.githubusercontent.com/46308407/185853110-0c0ad061-6d95-478e-8b06-0cb617214a91.png)

- The authors observe that the cross-attention map between text prompt and the intermediate diffusion features retain significant spatial information about the final output.

![image](https://user-images.githubusercontent.com/46308407/185853224-1126db69-4428-4172-b629-af223aec84aa.png)

- Hence, they suggest to manipulate the cross-attention map to edit the image based on the text prompt while preserving the non-canonical details (*e.g., overall scene arrangement, background*)
- 3 main operations are introduced: 1/ word swap 2/ phrase insertion 3/ attenuation (emphasis)

![image](https://user-images.githubusercontent.com/46308407/185853465-b6226f3b-04c4-4aca-8c38-62ad16e681f0.png)

- As diffusion models operate on coarse-to-fine scale, attention map injection can be temporarily controlled to yield better output.

![image](https://user-images.githubusercontent.com/46308407/185853512-d4326549-d4a6-422d-ba84-2a3d4d90eaf8.png)
![image](https://user-images.githubusercontent.com/46308407/185853813-e27e70ff-e879-4c00-b3fc-72f7b57f2c6d.png)
![image](https://user-images.githubusercontent.com/46308407/185853865-9dbc456c-a757-427b-a1e1-e247422eb3a0.png)
![image](https://user-images.githubusercontent.com/46308407/185853987-02a11daa-d87c-4aba-8c7c-571e34028e1c.png)

- In case of real image editing, employ deterministic DDIM to invert the image first, and apply similar procedure which doesn't always yield the best result.

![image](https://user-images.githubusercontent.com/46308407/185854129-7cc13098-31e0-4487-9a39-b3ed646555ac.png)

- It can leverage implicit mask (formed by attention map) for local editing.

![image](https://user-images.githubusercontent.com/46308407/185854208-ee10a95f-c545-42d7-ae80-6a9d188114a4.png)

# Evaluation

- Aforementioned.

# Inspiration

- Very interesting paper. It is well motivated, and the authors walk us through in a very straightforward yet gentle manner.
- It's interesting how directly operating on the cross-attention map enables fine control by the text prompt.
- What's still left: beyond object level representations. Relationship modeling and whole scene understanding is still not straightforward and subpar, in my personal opinion.
